### PR TITLE
Release update

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update the changelog, fix a few warnings, update the release date

### CODE REVIEW

**Comment by phlptp** [comment]
this will not get merged until after the connector MR is merged.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,40 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+## [3.5.0][] - 2024-02-05
+
+Major release including connector app capability, a refresh of the of the dataBuffer interface, and other bug fixes
+
+### Fixed
+
+- Fixed some issues regarding config files in the apps, where the config file was not being handled consistently
+- Fixed a few more sporadically failing testing cases and a rare bug in the use of `wait_for_current_time` flag when entering executing mode
+- Fixed a memory growth issue related to find operations in the C interface
+- resolve some issues with the threadSanitizer and other sporadic failures in the test cases.
```
Comment (nightlark) [inline CHANGELOG.md:21]:
```suggestion
- Resolved some issues with the threadSanitizer and other sporadic failures in the test cases
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,40 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+
+### Changed
+
+- Test with Boost 1.84 and CMake 3.28
+- large refactor of the dataBufferAPI, this is now considered stable, all dataBuffer related methods now start with helicsDataBuffer for consistency with other API methods and the XXXToBytes methods are changed to FillFromXXXX to be more consistent with other API's
```
Comment (nightlark) [inline CHANGELOG.md:26]:
```suggestion
- Large refactor of the dataBufferAPI, this is now considered stable, all dataBuffer related methods now start with helicsDataBuffer for consistency with other API methods and the XXXToBytes methods are changed to FillFromXXXX to be more consistent with other API's
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,40 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- Update 3rd party libraries
+- Update copyright to 2024
+- autobuild ZMQ now set to 4.3.5
```
Comment (nightlark) [inline CHANGELOG.md:29]:
```suggestion
- Autobuild ZMQ now set to 4.3.5
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,40 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+
+### Added
+
+- Added Connector app to enable automated connections of the federates via a set of matching rules, and allow optional interface generation via a query/command
+- Added helics::systemInfo() to the C++ shared library
+- Added ability generate translators from config files
```
Comment (nightlark) [inline CHANGELOG.md:35]:
```suggestion
- Added ability to generate translators from config files
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,40 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- Added API's to get and send dataBuffers from publication/input/endpoints
```
Comment (nightlark) [inline CHANGELOG.md:36]:
```suggestion
- Added API's to get and send dataBuffers from publications/inputs/endpoints
```

File: docs/ROADMAP.md [hunk 6]
```diff
@@ -2,14 +2,21 @@
...
-## \[3.5\] ~ Summer 2023
+## \[3.6\] ~ Summer 2024
 
 - Single thread cores
+- Update IPC core
 - Some of the other features listed below
+- This release will likely update HELICS to use C++20 and update minimum Compilers, CMake, boost, and other dependencies.
+  - GCC 11
+  - clang 14
+  - Cmake 3.22
```
Comment (nightlark) [inline docs/ROADMAP.md:13]:
```suggestion
  - CMake 3.22
```

File: docs/Public_API.md [hunk 7]
```diff
@@ -59,6 +58,7 @@ The public API includes the following
   - Tracer.hpp
   - Probe.hpp
   - Clone.hpp
+  - Connector.hpp  (considered in Beta and subject to change)
```
Comment (nightlark) [inline docs/Public_API.md:61]:
```suggestion
  - Connector.hpp (considered in Beta and subject to change)
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**